partition_gpu/partition_gpu.go

```diff
@@ -98,18 +107,27 @@ func main() {
 		os.Exit(1)
 	}
 	if !migModeEnabled {
+		glog.Infof("Checking the GPU type now.")
+		gpuType, err := checkGpuType()
+		if err != nil {
+			glog.Errorf("Failed to check GPU Type: %v", err)
+			os.Exit(1)
+		}
+		glog.Infof("Got GPU type used: %s", gpuType)
 		glog.Infof("MIG mode is not enabled. Enabling now.")
 		if err := enableMigMode(); err != nil {
 			glog.Errorf("Failed to enable MIG mode: %v", err)
 			os.Exit(1)
 		}
-		glog.Infof("Rebooting node to enable MIG mode")
-		if err := rebootNode(); err != nil {
-			glog.Errorf("Failed to trigger node reboot after enabling MIG mode: %v", err)
-		}
+		if gpuType == Nvidia40gbA100 || gpuType == Nvidia80gbA100 {
+			glog.Infof("Rebooting node to enable MIG mode")
+			if err := rebootNode(); err != nil {
+				glog.Errorf("Failed to trigger node reboot after enabling MIG mode: %v", err)
+			}
 
-		// Exit, since we cannot proceed until node has rebooted, for MIG changes to take effect.
-		os.Exit(1)
+			// Exit, since we cannot proceed until node has rebooted, for MIG changes to take effect.
```

Comment:
given the comment, might want to exit anyway is mig mode isnt enabled. So there should should be a a os.Exit(1) outside of this `if gpuType` clause

Reply:
os.Exit(1) is only needed for Nvidia40gbA100 and Nvidia80gbA100, @Jiaqicao257 please add more comments to explain this logic.

Reply:
Added comments to clarify this